From d3ecbca94f178716bcae957c8cd5501389d182aa Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 26 Aug 2014 12:49:29 -0400
Subject: [tools] turbostat: Drop temperature checks

Message-id: <1409057382-26432-19-git-send-email-prarit@redhat.com>
Patchwork-id: 89100
O-Subject: [RHEL7.1 PATCH BZ 1133893 18/31] tools / power: turbostat: Drop temperature checks
Bugzilla: 1133893
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1133893

commit 3482124a6a22c631df23958df497f000ba0e1667
Author: Jean Delvare <jdelvare@suse.de>
Date:   Thu May 1 11:40:19 2014 +0200

    tools / power: turbostat: Drop temperature checks

    The Intel 64 and IA-32 Architectures Software Developer's Manual says
    that TjMax is stored in bits 23:16 of MSR_TEMPERATURE TARGET (0x1a2).
    That's 8 bits, not 7, so it must be masked with 0xFF rather than 0x7F.

    The manual has no mention of which values should be considered valid,
    which kind of implies that they all are. Arbitrarily discarding values
    outside a specific range is wrong. The upper range check had to be
    fixed recently (commit 144b44b1) and the lower range check is just as
    wrong. See bug #75071:

    https://bugzilla.kernel.org/show_bug.cgi?id=75071

    There are many Xeon processor series with TjMax of 70, 71 or 80
    degrees Celsius, way below the arbitrary 85 degrees Celsius limit.
    There may be other (past or future) models with even lower limits.

    So drop this arbitrary check. The only value that would be clearly
    invalid is 0. Everything else should be accepted.

    After these changes, turbostat is aligned with what the coretemp
    driver does.

    Signed-off-by: Jean Delvare <jdelvare@suse.de>
    Cc: Len Brown <len.brown@intel.com>
    Acked-by: Guenter Roeck <linux@roeck-us.net>
    Reviewed-by: Josh Triplett <josh@joshtriplett.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Tony Camuso <tcamuso@redhat.com>
Cc: Dean Nelson <dnelson@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/power/x86/turbostat/turbostat.c b/tools/power/x86/turbostat/turbostat.c
index 1aea052..835caae 100644
--- a/tools/power/x86/turbostat/turbostat.c
+++ b/tools/power/x86/turbostat/turbostat.c
@@ -1971,13 +1971,13 @@ int set_temperature_target(struct thread_data *t, struct core_data *c, struct pk
  if (get_msr(0, MSR_IA32_TEMPERATURE_TARGET, &msr))
   goto guess;
 
- target_c_local = (msr >> 16) & 0x7F;
+ target_c_local = (msr >> 16) & 0xFF;
 
  if (verbose)
   fprintf(stderr, "cpu%d: MSR_IA32_TEMPERATURE_TARGET: 0x%08llx (%d C)\n",
    cpu, msr, target_c_local);
 
- if (target_c_local < 85 || target_c_local > 127)
+ if (!target_c_local)
   goto guess;
 
  tcc_activation_temp = target_c_local;
-- 
1.7.1